crates/iceberg/src/arrow/schema.rs

```diff
@@ -385,25 +389,236 @@ impl ArrowSchemaVisitor for ArrowSchemaConverter {
+    ) -> crate::Result<ArrowSchemaOrFieldOrType> {
+        match p {
+            crate::spec::PrimitiveType::Boolean => {
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Boolean))
+            }
+            crate::spec::PrimitiveType::Int => Ok(ArrowSchemaOrFieldOrType::Type(DataType::Int32)),
+            crate::spec::PrimitiveType::Long => Ok(ArrowSchemaOrFieldOrType::Type(DataType::Int64)),
+            crate::spec::PrimitiveType::Float => {
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Float32))
+            }
+            crate::spec::PrimitiveType::Double => {
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Float64))
+            }
+            crate::spec::PrimitiveType::Decimal { precision, scale } => {
+                let (precision, scale) = {
+                    let precision: u8 = precision.to_owned().try_into().map_err(|err| {
+                        Error::new(
+                            crate::ErrorKind::DataInvalid,
+                            "incompatible precision for decimal type convert",
+                        )
+                        .with_source(err)
+                    })?;
+                    let scale = scale.to_owned().try_into().map_err(|err| {
+                        Error::new(
+                            crate::ErrorKind::DataInvalid,
+                            "incompatible scale for decimal type convert",
+                        )
+                        .with_source(err)
+                    })?;
+                    (precision, scale)
+                };
+                validate_decimal_precision_and_scale::<Decimal128Type>(precision, scale).map_err(
+                    |err| {
+                        Error::new(
+                            crate::ErrorKind::DataInvalid,
+                            "incompatible precision and scale for decimal type convert",
+                        )
+                        .with_source(err)
+                    },
+                )?;
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Decimal128(
+                    precision, scale,
+                )))
+            }
+            crate::spec::PrimitiveType::Date => {
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Date32))
+            }
+            crate::spec::PrimitiveType::Time => Ok(ArrowSchemaOrFieldOrType::Type(
+                DataType::Time32(TimeUnit::Microsecond),
+            )),
+            crate::spec::PrimitiveType::Timestamp => Ok(ArrowSchemaOrFieldOrType::Type(
+                DataType::Timestamp(TimeUnit::Microsecond, None),
+            )),
+            crate::spec::PrimitiveType::Timestamptz => Ok(ArrowSchemaOrFieldOrType::Type(
+                // Timestampz always stored as UTC
+                DataType::Timestamp(TimeUnit::Microsecond, Some("+00:00".into())),
+            )),
+            crate::spec::PrimitiveType::String => {
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Utf8))
```

Comment:
nit: It is just an alias, but makes it easier to read:
```suggestion
                Ok(ArrowSchemaOrFieldOrType::Type(DataType::String))
```

Reply:
This is [DataType defined in arrow](https://docs.rs/arrow/latest/arrow/datatypes/enum.DataType.html) and we can't change it.

Reply:
+1, there is no string data type in arrow.